flags/state.py

```diff
@@ -1,13 +1,31 @@
+from __future__ import absolute_import
+
+from flags.middleware import FlagConditionsMiddleware
 from flags.settings import get_flags
 
 
 def flag_state(flag_name, **kwargs):
     """ Return the value for the flag by passing kwargs to its conditions """
-    try:
-        return get_flags()[flag_name].check_state(**kwargs)
-    except KeyError:
+    flags = None
+
+    # If the request is given as a kwargs, and the FlagConditionsMiddleware is
+    # enabled, use the flag conditions attached to the request.
+    if 'request' in kwargs:
+        flags = getattr(
+            kwargs['request'],
+            FlagConditionsMiddleware.request_attribute,
+            None
+        )
+
+    if flags is None:
+        flags = get_flags()
+
+    if flag_name not in flags:
```

Comment:
Rather than catching a `KeyError` here, I've decided to check if the `flag_name` is in the flags we have, period. I think that makes intent a lot clearer.

Reply:
Doing both `in` and `[]` means that you're doing the same lookup twice. And while it should always be the case that if `get_flags()` returns a dict that `in` means that `[]` will always succeed, maybe there's some weird case where it might not?

As an alternative, what about:

```py
flag = flags.get(flag_name)

# If the flag isn't defined, its state is considered False.
if flag is None:
    return False
```

As an aside, and if you're planning a breaking change, what would you think about modifying the API so that this method returns `None` if the flag is not defined? That shouldn't affect things downstream but would enable clients to test for flag definition should there be a use case for that.

Reply:
`None` is a good idea, since it's still falsey.